subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -697,6 +700,50 @@ static bool valid_register_param(const struct bt_csip_set_member_register_param
 	return true;
 }
 
+static void remove_csis_char(const struct bt_uuid *uuid, struct bt_gatt_service *svc)
+{
+	size_t attrs_to_rem;
+
+	/* Rank does not have any CCCD */
+	if (bt_uuid_cmp(uuid, BT_UUID_CSIS_RANK) == 0) {
+		attrs_to_rem = CSIS_RANK_CHAR_ATTR_COUNT;
+	} else {
+		attrs_to_rem = CSIS_CHAR_ATTR_COUNT;
+	}
+
+	/* Start at index 4 as the first 4 attributes are mandatory */
+	for (size_t i = 4U; i < svc->attr_count; i++) {
+		if (bt_uuid_cmp(svc->attrs[i].uuid, uuid) == 0) {
+			/* Remove the characteristic declaration, the characteristic value and
+			 * potentially the CCCD. The value declaration will be a i - 1, the
+			 * characteristic value at i and the CCCD is potentially at i + 1
+			 */
+
+			/* We use attrs_to_rem to determine whether there is a CCCD after the
+			 * characteristic value or not, which then determines if this is the last
+			 * characteristic or not
+			 */
+			if (i == (svc->attr_count - (attrs_to_rem - 1))) {
+				/* This is the last characteristic in the service: just decrement
+				 * the attr_count by number of attributes to remove
+				 * (CSIS_CHAR_ATTR_COUNT)
+				 */
+			} else {
+				/* Move all following attributes 3 locations "up" */
+				for (size_t j = i - 1U; j < svc->attr_count - attrs_to_rem; j++) {
```

Comment:
My comment may be unclear :D 

For the sake of the argument, let's assume that we just have an array of services and characteristics like
```
[0] = Primary service
[1] = Characteristic A
[2] = Characteristic B
[3] = Characteristic C
[4] = Characteristic D
```

And we want to remove `Characteristic B`. 
What this does, is to move everything from (and including) `Characteristic C` up, so this becomes

```
[0] = Primary service
[1] = Characteristic A
[2] = Characteristic C
[3] = Characteristic D
```

Technically `[4]` is also `Characteristic D`, but since we decrement the `attr_count`, it can ignored. 

The term "up" may be confusing, and perhaps we should say something like "reduce the index of" or similar. 

The changes done here would also have been much simpler if the GATT API used a a linked list for the attributes :D

Reply:
It's my bad, I see how I read it wrong now, lgtm